Extraction of i18n strings in system panel

greenwoodt opened the pull request:
<!--
NOTE: We are in the middle of a big redesign of the frontend.
That could mean that your PR will not get through on the next few months.
Please see https://github.com/decidim/decidim/discussions/9512
-->

#### :tophat: What? Why?
Extracting and adding i18n translations to system docs where needed, making the use of strings not needed. 

#### :pushpin: Related Issues
*Link your PR to an issue*
- Related to #?
- Fixes #11481 

#### Testing


### :camera: Screenshots

:hearts: Thank you!

alecslupu commented:
@greenwoodt this pr is not complete, as there are more strings to be extracted. Please see all the comments in the ticket

greenwoodt commented:
> @greenwoodt this pr is not complete, as there are more strings to be extracted. Please see all the comments in the ticket

I'm fully aware @alecslupu thanks. I'll notify you for a review once it is.

greenwoodt commented:
@alecslupu I have changed and added translations according to your request on issue: , however there was one I was stuck with on adding as the string does interpolate a class variable, located here: ```decidim-system/app/views/decidim/system/devise/shared/_links.html.erb```on line 23. 

I wondered whether you could provide insight on changing this? Does one at the interploation on the locales.en file directly and just call the translation?

greenwoodt commented:
I have reverted line 12 back to ```Decidim``` here: ```decidim-system/app/views/layouts/decidim/system/login.html.erb```as it was failing tests to do with yield and not being able to find the translation.

andreslucena commented:
> I have reverted line 12 back to `Decidim` here: `decidim-system/app/views/layouts/decidim/system/login.html.erb`as it was failing tests to do with yield and not being able to find the translation.

What if we add the string that's missing in the en.yml file? I'm referring to this one: 

>  Translation missing: en.layouts.decidim.shared.layout_center.decidim

greenwoodt commented:
> > I have reverted line 12 back to `Decidim` here: `decidim-system/app/views/layouts/decidim/system/login.html.erb`as it was failing tests to do with yield and not being able to find the translation.
> 
> What if we add the string that's missing in the en.yml file? I'm referring to this one:
> 
> > Translation missing: en.layouts.decidim.shared.layout_center.decidim

The file still fails a test, even after adding the string manually. I can try again today and let you know.

andreslucena commented on decidim-system/app/views/decidim/system/devise/shared/_links.html.erb:
```diff
@@ -3,23 +3,23 @@
+  <%= link_to t(".did_not_receive_unlock_instructions?"), new_unlock_path(resource_name) %>
 <% end -%>
 
 <%- if devise_mapping.omniauthable? %>
```
I think I wasn't clear in my comment, but I meant to directly delete these lines (from 21 to 25) as they are not used. Can you do that @greenwoodt ? Thanks

greenwoodt replied:
I understood Andres. Will have it completed.

alecslupu commented on decidim-system/config/locales/en.yml:
```diff
@@ -67,6 +70,24 @@ en:
+            forgot_your_password: Forgot your password
+            send_me_reset_password_instructions: Send me reset password instructions
+        sessions:
+          new:
+            log in: Log in
+            log_in: Log in
```
```suggestion
          new:
            log_in: Log in
```

alecslupu commented on decidim-system/app/views/decidim/system/devise/sessions/new.html.erb:
```diff
@@ -12,7 +12,7 @@
 
   <div class="form__wrapper-block">
-    <%= f.submit "Log in", class: "button button__sm md:button__lg md:button__lg button__secondary" %>
+    <%= f.submit t(".log in"), class: "button button__sm md:button__lg md:button__lg button__secondary" %>
```
```suggestion
    <%= f.submit t(".log_in"), class: "button button__sm md:button__lg md:button__lg button__secondary" %>
```

alecslupu reviewed (changes requested):
There is one key that should not be as it is.

alecslupu commented on decidim-system/app/views/decidim/system/devise/passwords/edit.html.erb:
```diff
@@ -1,12 +1,12 @@
   <div class="form__wrapper">
     <% if @minimum_password_length %>
-      <em>(<%= @minimum_password_length %> characters minimum)</em><br>
+      <em>(<%= @minimum_password_length %> <%= t(".characters minimum") %> )</em><br>
```
```suggestion
      <em>(<%= @minimum_password_length %> <%= t(".minimum_characters") %> )</em><br>
```

alecslupu commented on decidim-system/config/locales/en.yml:
```diff
@@ -67,6 +70,24 @@ en:
+        passwords:
+          edit:
+            change_your_password: Change your password
+            characters minimum: Characters minimum
```
```suggestion
            minimum_characters: Characters minimum
```

alecslupu reviewed (changes requested):
@greenwoodt , please check also : decidim-system/app/views/decidim/system/devise/shared/_links.html.erb  Line 2 ( there is a "Log in" left befind) 

https://github.com/decidim/decidim/pull/11478#discussion_r1293351790

greenwoodt commented:
> @greenwoodt , please check also : decidim-system/app/views/decidim/system/devise/shared/_links.html.erb Line 2 ( there is a "Log in" left befind)
> 
> [#11478 (comment)](https://github.com/decidim/decidim/pull/11478#discussion_r1293351790)

Sadly this change isn't allowing the Main folder tests to pass: 

![Screenshot 2023-09-19 at 09 54 28](https://github.com/decidim/decidim/assets/101816158/a24c9ac3-ef47-423d-8a40-e09a2ad7ce67)

Yet even with a i18n key, I'm being told one is missing: 

<img width="910" alt="Screenshot 2023-09-19 at 09 16 14" src="https://github.com/decidim/decidim/assets/101816158/2b55b562-7d7f-4bd1-bd33-ce7a664b2e0e">

Any work around to this you can think of @alecslupu @andreslucena ??

andreslucena reviewed (approved):
I've just checked out all the comments from the original issue and I found a missing one: 
https://github.com/decidim/decidim/pull/11478#discussion_r1293365081

Regarding the rest, codewise is 🆗

andreslucena reviewed (dismissed):
Sorry, wrong button 😅 
Can you check [my last review](https://github.com/decidim/decidim/pull/11555#pullrequestreview-1637021989)? 🙏🏽

greenwoodt commented:
> Sorry, wrong button 😅 Can you check [my last review](https://github.com/decidim/decidim/pull/11555#pullrequestreview-1637021989)? 🙏🏽

No worries! Check my comment [here](https://github.com/decidim/decidim/pull/11555#issuecomment-1721349930) regard this change that didn't work. I left it out purposefully as it was causing errors to the tests.

alecslupu commented:
> > Sorry, wrong button 😅 Can you check [my last review](https://github.com/decidim/decidim/pull/11555#pullrequestreview-1637021989)? 🙏🏽
> 
> No worries! Check my comment [here](https://github.com/decidim/decidim/pull/11555#issuecomment-1721349930) regard this change that didn't work. I left it out purposefully as it was causing errors to the tests.

I have checked [66fc58d](https://github.com/decidim/decidim/pull/11555/commits/66fc58d2a7fb5c436ee700c684781bf260f5c551), and it seems that you have forgot to add it to en.yml file ... it should work without any kind of issues

greenwoodt commented:
@alecslupu @andreslucena updated and ready.

alecslupu reviewed (approved):
LGTM
